docs/ldml/tr35.html

```diff
@@ -1267,21 +1277,30 @@ <h5><a name="Language_Tag_to_Locale_Identifier" href=
     converted to a valid Unicode BCP 47 locale identifier by
     performing the following transformation.</p>
     <ol>
-	  <li>Canonicalize the syntax of the language tag (afterwards, there will be
-	  no extlang subtags) as per <a href="#Canonical_Unicode_Locale_Identifiers">3.2.1 Canonical Unicode Locale Identifiers</a>.</li>
-      <li>If the BCP 47 primary language subtag matches the
-      <i>type</i> attribute of a <i>languageAlias</i> element in
-      <a href="tr35-info.html#Supplemental_Data">Supplemental
-      Data</a>, replace the language subtag with the
-      <i>replacement</i> value.
-        <ol>
-          <li>If there are additional subtags in the
-          <i>replacement</i> value, add them to the result, but
-          only if there is no corresponding subtag already in the
-          tag.</li>
-          <li>Five special deprecated grandfathered codes (such as <em>i-default</em>) are in  type attributes, and are also replaced.</li>
-          <li><strong>Note: </strong>there are currently no valid 4-letter primary language subtags. While it is extremely unlikely that BCP47 would ever register them, if so then <i>languageAlias</i> mappings will be supplied for them, mapping to defined CLDR language subtags (from the idStatus=&quot;reserved&quot; set).</li>
-        </ol>
+      <li>Canonicalize the syntax of the language tag (afterwards, there will
```

Comment:
Hmmm. We have some very similar processing here. I think we need to make this a "subroutine".